ContactsMRC/ContactsMRC/ViewControllers/LSIContactsViewController.m

```diff
@@ -0,0 +1,98 @@
+//
+//  LSIContactsViewController.m
+//  ContactsMRC
+//
+//  Created by Bhawnish Kumar on 6/17/20.
+//  Copyright © 2020 Bhawnish Kumar. All rights reserved.
+//
+
+#import "LSIContactsViewController.h"
+#import "LSIContact.h"
+@interface LSIContactsViewController ()
+
+@property (retain, nonatomic) IBOutlet UITextField *nameTextField;
+@property (retain, nonatomic) IBOutlet UITextField *emailAddressTextField;
+@property (retain, nonatomic) IBOutlet UITextField *telephoneTextField;
+@property (retain, nonatomic) IBOutlet UIBarButtonItem *editButton;
+@property (retain, nonatomic) IBOutlet UIButton *saveButton;
+
+@property (nonatomic)BOOL contactUpdated;
+
+@end
+
+@implementation LSIContactsViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    if (self.contact == nil) {
+        self.contactUpdated = NO;
+        [self.editButton setEnabled:NO];
+        
+    } else {
+        [self.editButton setEnabled:YES];
+        self.contactUpdated = YES;
+        [self updateViews];
+    }
+    // Do any additional setup after loading the view.
+}
+
+- (void)updateViews {
+    if (self.contactUpdated == YES) {
+        [self.nameTextField setUserInteractionEnabled:self.isEditing];
+        [self.emailAddressTextField setUserInteractionEnabled:self.isEditing];
+        [self.telephoneTextField setUserInteractionEnabled:self.isEditing];
+        
+        self.nameTextField.text = self.contact.name;
+        self.emailAddressTextField.text = self.contact.emailAddress;
+        self.telephoneTextField.text = self.contact.telephone;
+    }
+}
```

Comment:
```objc
- (void)updateViews {
    if (self.contactUpdated == YES) {
        
        // I like the idea, but there's no visual indication that this is not
        // editable, so I kept tapping on it, and thought the app was broken
        
        // Instead of disabling, make a editiable version of the detailView
        // that is distinct from the viewing experience like in Contacts
        [self.nameTextField setUserInteractionEnabled:self.isEditing];
        [self.emailAddressTextField setUserInteractionEnabled:self.isEditing];
        [self.telephoneTextField setUserInteractionEnabled:self.isEditing];
        
        self.nameTextField.text = self.contact.name;
        self.emailAddressTextField.text = self.contact.emailAddress;
        self.telephoneTextField.text = self.contact.telephone;
    }
}

```